### PR TITLE
Add "pharos license assign" command

### LINKED ISSUE #667: License assign sub-command
Should exchange a valid license key to a subscription token and store it to `pharos-license` secret in `kube-system` namespace. 

This replaces `license-manager` addon functionality.

### PR DESCRIPTION
**Pull request by kke**
Fixes #667 

```console
$ pharos license assign -c cluster.yml
Enter Kontena Pharos license key: xxx-xxxx-x-xx
==> Reading instructions ...
    Exchanging license key for a subscription token
    Added subscription token to pharos cluster secrets
```

### CODE REVIEW

File: lib/pharos/license_assign_command.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    def execute
+      signal_usage_error 'invalid LICENSE_KEY format' unless license_key.match?(/^\h{8}-(?:\h{4}-){3}\h{12}$/)
+      retry_times = 0
+      ssh.exec!("kubectl create secret generic pharos-cluster --namespace=kube-system --from-literal=key=#{subscription_token.shellescape}")
+      logger.info "Added subscription token to pharos cluster secrets"
+    rescue Pharos::SSH::RemoteCommand::ExecError
+      retry_times += 1
+      raise if retry_times > 1
+      ssh.exec!("kubectl delete secret pharos-cluster --namespace=kube-system")
```
Comment (jakolehm) [inline lib/pharos/license_assign_command.rb]: Why do delete on rescue?

File: lib/pharos/license_assign_command.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      @ssh ||= Pharos::SSH::Manager.new.client_for(master_host)
+    end
+
+    def master_host
+      @master_host ||= config.master_hosts.first
+    end
+
+    def subscription_token_request
+      logger.info "Exchanging license key for a subscription token"
+      http_client.post(
```
[the remark this answers is not in the document]
Comment (kke) [reply lib/pharos/license_assign_command.rb]: image

[the remark this answers is not in the document]
Comment (jakolehm) [reply lib/pharos/license_assign_command.rb]: Ok, didn't notice the error handling.. all good 😄

**Comment by jakolehm** [comment]
@nevalla PTAL

**Comment by kke** [comment]
> Just thinking should we also prompt the description instead of using the default?

In that case it should only be prompted when also the key is prompted.

File: lib/pharos/root_command.rb [hunk 3]
```diff
@@ -4,14 +4,16 @@
...
 
 module Pharos
   class RootCommand < Pharos::Command
     banner "#{File.basename($PROGRAM_NAME)} - Kontena Pharos cluster manager"
 
     subcommand ["build", "up"], "initialize/upgrade cluster", UpCommand
     subcommand "kubeconfig", "fetch admin kubeconfig file", KubeconfigCommand
-    subcommand ["reset"], "reset cluster", ResetCommand
+    subcommand "reset", "reset cluster", ResetCommand
+    subcommand "license", "license subcommands", LicenseCommand
```
Comment (jakolehm) [inline lib/pharos/root_command.rb]: This command should not be part of the "oss" build.

**Comment by kke** [comment]
Moved to non-oss.

File: lib/pharos/root_command.rb [hunk 4]
```diff
@@ -22,3 +22,5 @@ def self.run
     end
   end
 end
+
+Dir.glob(File.join(__dir__, '..', '..', 'non-oss', 'commands', '*.rb')).each { |non_oss_command| require non_oss_command }
```
Comment (jakolehm) [inline lib/pharos/root_command.rb]: We probably could load `non-oss/setup.rb` or similar that handles all the commands/addons. But that kind of change is out of scope of this pr.
